text/0017-message/0017-message.md

```diff
@@ -0,0 +1,169 @@
+        </tr>
+    <tr>
+        <td> Parents' length </td>
+        <td> uint8</td>
+        <td> The number of messages we directly approve. Can be any value between 1-8.</td>
+    </tr>
+    <tr>
+        <td>Parents </td>
+        <td>ByteArray[32 * `parents length`]</td>
+        <td>The Message IDs that are referenced.</td>
+    </tr>
+    <tr>
+        <td>Payload Length</td>
+        <td>uint32</td>
+        <td> The length of the Payload. Since its type may be unknown to the node it must be declared in advanced. 0 length means no payload will be attached.</td>
+    </tr>
+    <tr>
+        <td colspan="1">
+            Payload
+        </td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Generic Payload</summary>
+                <blockquote>
+                An outline of a general payload
+                </blockquote>
+                <table>
+                    <tr>
+                        <th>Name</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>Payload Type</td>
+                        <td>uint32</td>
+                        <td>
+                            The type of the payload. It will instruct the node how to parse the fields that follow.
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Data Fields</td>
+                        <td>ANY</td>
+                        <td>A sequence of fields, where the structure depends on <code>payload type</code>.</td>
+                    </tr>
+                </table>
+            </details>
+    <tr>
+        <td>Nonce</td>
+        <td>uint64</td>
+        <td>The nonce which lets this message fulfill the Proof-of-Work requirement.</td>
+    </tr>
+</table>
+
+### Message validation
+
+A message is considered valid, if the following syntactic rules are met:
+
+1. The message size must not exceed 32 KiB (32 * 1024 bytes).
+2. When parsing the message is complete, there should not be any trailing bytes left that were not parsed.
+3. If the `payload type` is known to the node.
+4. If the [Message PoW Hash](https://github.com/Wollac/protocol-rfcs/blob/message-pow/text/0024-message-pow/0024-message-pow.md) will contain at least the number of trailing 0 trits the node defines as required.
+5. `Parents' length` must be between 1-8.
+
+
+### Payloads
+
+A message may contain a payload. The specification of the payloads is out of scope of this RFC. Below is a table of the currently specified core payloads with a link to their specifications. The `indexation payload` will be specified here as an example:
```

Comment:
Here it say it "may" contain a payload, what would the message format be if no payload exists? It seems like by default the rest of the message would be zero'd out which could be interpreted as a Signed Transaction (according to the table below).